### ISSUE TITLE
External Components are not loaded into smac child processes and external components subclasess

### ISSUE THREAD

**Issue by hmendozap** [automl/auto-sklearn#56]
After loading an external component feature as directed in the  [manual](http://auto-sklearn.readthedocs.org/en/development/extending.html#extending) with:

``` python
from component import DeepFeedNet
add_classifier(DeepFeedNet.DeepFeedNet)
autosk.AutoML(include_estimators=['DeepFeedNet']))
```

In the main process it does not generate error or warning, but after reviewing the smac logs, this error is reported:
[ Shortened... ]
16:57:30.806 [CLI TAE (STDERR Thread - #0)] WARN  c.u.c.b.a.t.b.c.CommandLineAlgorithmRun - [PROCESS-ERR](hp_value, hyperparameter))
16:57:30.807 [CLI TAE (STDERR Thread - #0)] WARN  c.u.c.b.a.t.b.c.CommandLineAlgorithmRun - [PROCESS-ERR]  ValueError: Hyperparameter instantiation 'DeepFeedNet' is illegal for hyperparameter classifier:**choice**, Type: Categorical, Choices: {adaboost, bernoulli_nb, decision_tree, extra_trees, gaussian_nb, gradient_boosting, k_nearest_neighbors, lda, liblinear_svc, libsvm_svc, multinomial_nb, passive_aggressive, proj_logit, qda, random_forest, sgd}, Default: random_forest

Which after further review the error is (presumably) at the loading of the classifiers in configuration of datamanager in the subprocesses:

``` python
 193       cs = get_configuration_space(D.info)
 194       configuration = configuration_space.Configuration(cs, params)
```

**Comment by hmendozap** [comment]
Adding to the previous comment, a subclass from an external component it must have multiple-inheritance from both the component and AutoSklearnAlgorithm (eg. ```class ConstrainedNeuralNet(NeuralNet, AutoSklearnClassificationAlgorithm): [...] ```), is that a desired feature to have?

**Comment by mfeurer** [comment]
Interesting, just subclassing an existing component isn't enough?

**Comment by hmendozap** [comment]
It's not enough from what I tested and I believe is because the base class is compared against the immediate superclass and not the mro or classes tree...

**Comment by mfeurer** [comment]
Okay, I'm not sure if this is still a relevant issue, but I fixed it with commit 78154d7.
